ci.md

```diff
@@ -0,0 +1,232 @@
+    * **Actions**: When there are code check failures, the build is marked red ❌.
+      On the build logs, the checker failures should be under the `Code check`
+      collapsible.
+* There are **infrastructure failures**. Contributors should leave a comment on
+    the pull request for maintainers, who will report these to the build farmer.
+    * **Jenkins**: When there's an infra failure, the build is marked red 🔴.
+      These can manifest in various ways. In general, if the build logs don't
+      fall in the categories above, there's a high chance it's an infrastructure
+      failure. Common patterns usually have several `java` or `hudson` messages.
+    * **Actions**: Infrastructure failures haven't been identified.
+
+## Triggering CI
+
+CI is usually triggered automatically when code is pushed or a pull request is
+opened. But often it's necessary to re-trigger CI without making changes to the
+code if:
+
+* There were infrastructure failures in a previous build.
+* There are fixes upstream, on infra or dependencies, that may make a new build pass.
+
+The following methods can be used to re-trigger builds:
+
+* **Jenkins**
+    * Make a comment starting with `@osrf-build run tests` on the pull request
+      and all Jenkins builds will be re-triggered.
+    * To restart just one specific build and avoid re-running builds that aren't
+      necessary, go to the failing build and click `Retry`. This is only available
+      to maintainers.
+* **Actions**
+    * On the top-right of a build, click `Re-run jobs`. This button is only
+      available to maintainers and sometimes mysteriously disappears.
+
+### Custom branches
+
+Maintainers can manually trigger builds for custom branches without opening pull
+requests, optionally building dependencies from source and changing other
+configurations for testing.
+
+One possible reason to trigger CI with custom dependency branches is testing
+changes across repositories. For example, if your `ign-gazebo` pull request
+needs a change from `ign-physics`, you could build a custom branch of `ign-physics`
+from source to check it works with `ign-gazebo` on CI before even opening an
+`ign-physics` pull request, or making a release.
+
+
+#### Actions
+
+By default, builds are run for every commit based on configuration inside the
+`.github/ci` directory. So it's possible to change the contents of that
+directory and immediately see updated CI results when you push commits.
+
+See the documentation on
+[action-ignition-ci](https://github.com/ignition-tooling/action-ignition-ci)
+to see what can be configured and how.
+
+Be sure to revert these changes before merging the pull request.
+
+#### Jenkins
+
+The `pr_any` jobs can be triggered manually for any branch by maintainers.
+Go to the job's page, for example
+[ignition_launch-ci-pr_any-ubuntu_auto-amd64](https://build.osrfoundation.org/job/ignition_launch-ci-pr_any-ubuntu_auto-amd64/),
+click on `Build with Parameters` to trigger a new build.
```

Comment:
I would leave a warning that this sometimes causes problems and link to https://github.com/ignition-tooling/release-tools/issues/242